test/integration/02-user-wallet/02-index.spec.ts

```diff
@@ -97,24 +96,22 @@ describe("UserWallet", () => {
 
     it("allows set username", async () => {
       let result = await setUsername({ username: "user0", id: accountId0 })
-      expect(!!result).toBeTruthy()
+      expect(result).not.toBeInstanceOf(Error)
       result = await setUsername({ username: "user1", id: accountId1 })
-      expect(!!result).toBeTruthy()
+      expect(result).not.toBeInstanceOf(Error)
     })
 
     it("does not allow set username if already taken", async () => {
       const username = "user0"
-
-      await createUserWallet(2)
       await expect(setUsername({ username, id: accountId2 })).resolves.toBeInstanceOf(
-        UsernameNotAvailableError,
+        AccountError,
```

Comment:
please use the specific error. The same applies for other changes in this file

Reply:
the specific error is path dependant, (depending on why tests have been run, one error (UsernameNotAvailableError ) or another one (UsernameIsImmutableError) might be returned which is why I return the parent common class.

Reply:
this is a common mistake in tests, if we are testing a case for specific error then we need to use the proper instance of the error to guarantee that the code is validating the correct path

Reply:
If this test is non-deterministic then there is something wrong with the setup.
Tests should be idempotent and deterministic.
What is preventing you from making this test order independent?

Reply:
because you had to test one condition first, and then the next one: https://github.com/GaloyMoney/galoy/blob/1807822e46e5b34722c0868c80bace6a6ba53ea5/src/app/accounts/set-username.ts#L20

Reply:
so if username is already set you won't go further.

Reply:
I'm not sure I understand what is preventing you from weighting this pseudocode:

test1:
```
user = getNewUser();
setUsername(user.id, "name");
expect(setUsername(user.id, "name")).toThrow(UsernameIsImmutableError()
```
test2:
```
user=getNewUser();
setUsername(user.id, ("name"));
user2=getNewUser();
expect(setUsername(userd.id, "name")).toThrow(UsernameNotAvailable)()
```